src/dataflow-types/src/client/controller/compute.rs

```diff
@@ -587,6 +613,16 @@ where
             self.update_read_capabilities(&mut read_capability_changes)
                 .await?;
         }
+
+        let storage_updates: Vec<_> = updates
+            .iter()
+            .filter(|(id, _)| self.storage_mut().collection(*id).is_ok())
+            .cloned()
+            .collect();
+        self.storage_mut()
+            .update_write_frontiers(&storage_updates)
+            .await?;
```

Comment:
This tells the storage controller about new uppers for collections targeted by sinks. @aljoscha has proposed that we should instead make the compute controller query the persist shards for their uppers directly.